rfcs/0039-minaform-tool-design.md

```diff
@@ -0,0 +1,117 @@
+
+### Minaform Help
+
+The help page will link to this RFC as the initial docs and provide a list of all flags, commands, etc and their usage. Some flags like `--auto-approve` or `--force` are directly passed to terraform and the documentation for those flags should also be provided in the help for minaform. 
+
+## Beyond the minaform tool
+
+### Prod and QA templates
+
+To properly provide consistent systems for key management and how to generate ledgers for a given network, the templates for production and qa networks will need to understand how to delegate stake to community members, where to download + source community member public keys, etc. As this will get complicated, for networks that use community member keys there may need to be a manual process interacting with `coda-network` or the existing hacky script in the `coda-network-integration` branch for that purpose. The priorities for key management are 1) flexible enough to manually create complex ledgers for specific networks, 2) straightforward and simple to deploy a network where we control all of the keys.
+
+These templates are specifically designed to handle the #1 pain point and more clearly outline what a "normal" qa net should look like, as well as sharing as much infrastructure and process as possible with the production environment. The production template and associated files should include a README.md with exactly how this template becomes a production network deployment, especially if the key management or other pieces are not completely automated.
+
+### Minaform Docker image
+
+To address pain points #7 and #8, and to some extent the larger problem of how to make it easy for anyone, from anywhere, to deploy networks, we need a minaform docker image that can be bundled alongside all of its dependencies (nodejs for coda-network, some of our deb packages like generate-keypair, terraform, helm, most of mina-automation repo, and the helm charts themselves). These dependencies can then be handled by updates to the docker image, and anyone having trouble with version mismatches can mount their credentials into a docker container and get moving quickly.
+
+In addition to use on any developer laptop, the docker image could be deployed directly into the cluster, google cloud shell, or a google cloud VM to allow for low-latency deployment to gcloud even from a high-latency network (hope that helps @bkase @mrmr1993 !).
+
+## Drawbacks
+[drawbacks]: #drawbacks
+
+Generating genesis proofs on developer laptops means downloading snark keys from s3 or large dockerfiles from docker hub, and can be difficult on a poor connection. I still believe this is worth the effort/time compared to the cost of generating the same data on each node for hundreds of nodes, and will make it very clear which genesis ledger is in use on a given network.
```

Comment:
A GCE instance for generating docker containers would help here. Might be too much work initially, but I know it would help me personally (as someone with rather bad internet).